internal/fs/fs_test.go

```diff
@@ -172,6 +173,85 @@ func TestRenameWithFallback(t *testing.T) {
 	}
 }
 
+func TestIsCaseSensitiveFilesystem(t *testing.T) {
+	if runtime.GOOS != "windows" && runtime.GOOS != "linux" {
+		t.Skip("skip this test on non-Windows, non-Linux")
+	}
```

Comment:
`darwin`? We run macOS specific tests on travis and by default macOS is also case insensitive, like windows.

Reply:
There are many OS recognised by Go runtime - https://github.com/golang/go/blob/master/src/go/build/syslist.go#L7. I thought that since I wasn't sure about the rest of them, it would be sufficient to test for 1 case sensitive and 1 case insensitive system. But I agree that it makes sense to check for macOS as it is among the most popular.